timemachine/md/builders.py

```diff
@@ -1,25 +1,90 @@
 import os
-from typing import Union
+from typing import List, Optional, Union
 
 import numpy as np
-from openmm import Vec3, app, unit
+from numpy.typing import NDArray
+from openmm import app, unit
+from rdkit import Chem
 
+from timemachine.fe.utils import get_romol_conf
 from timemachine.ff import sanitize_water_ff
+from timemachine.potentials.jax_utils import idxs_within_cutoff
 
 
-def strip_units(coords):
+def strip_units(coords) -> NDArray[np.float64]:
     return np.array(coords.value_in_unit_system(unit.md_unit_system))
 
 
-def build_protein_system(host_pdbfile: Union[app.PDBFile, str], protein_ff: str, water_ff: str):
+def remove_clashy_waters(
+    modeller: app.Modeller,
+    host_coords: NDArray[np.float64],
+    box: NDArray[np.float64],
+    water_idxs: NDArray,
+    mols: List[Chem.Mol],
+    clash_distance: float = 0.4,
+):
+    """Remove waters from an OpenMM modeler that clash with a set of molecules
+
+    Parameters
+    ----------
+    modeller: app.Modeller
+        Modeller to update in place
+
+    host_coords: NDArray[np.float64]
+        Coordinates of host, may be different than modeller.positions
+
+    box: NDArray[np.float64]
+        Box to evaluate PBCs under
+
+    water_idxs: NDArray[int]
+        The water idxs to consider for deletion
+
+    mols: List[Mol]
+        List of molecules to determine which waters are clashy
+
+    clash_distance: float
+        Distance from a ligand atom to a water atom to consider as a clash, in nanometers
+    """
+    water_coords = host_coords[water_idxs]
+    ligand_coords = np.concatenate([get_romol_conf(mol) for mol in mols])
+    clashy_idxs = idxs_within_cutoff(water_coords, ligand_coords, box, cutoff=clash_distance)
+    if len(clashy_idxs) == 0:
+        return
+    # Offset the clashy idxs with the first atom idx, else could be pointing at non-water atoms
+    clashy_idxs += np.min(water_idxs)
```

Comment:
Does this assume that `water_idxs` contains the indices of all of the water atoms in the solvent? If so, might note this explicitly in the docstring.

Reply:
It does will clarify.

Reply:
Clarified in https://github.com/proteneer/timemachine/pull/1339/commits/957775ecd2ab68f613f679a880bdcbea47f5bdac